File: week_1/day_1_arrays/move_zero_follow_up_questions/What_if_the_array_is_very_large.js
function moveZeroesLargeArray(nums) {
    let write = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] !== 0) {
            nums[write] = nums[i];
            write++;
        }
    }
    for (let i = write; i < nums.length; i++) {
        nums[i] = 0;
    }
}

// Example usage:
let arr = [0, 1, 0, 3, 12];
moveZeroesLargeArray(arr);
console.log(arr); // Output: [1, 3, 12, 0, 0]

File: week_1/day_1_arrays/move_zero_follow_up_questions/What_if_the_array_is_very_large.py
def moveZeroesLargeArray(nums):
    write = 0
    for num in nums:
        if num != 0:
            nums[write] = num
            write += 1
    for i in range(write, len(nums)):
        nums[i] = 0



nums = [0, 1, 0, 3, 12]
moveZeroesLargeArray(nums)
print(nums)
